testrunner-running/src/main/scala/edu/illinois/cs/statecapture/StateCapture.java

```diff
@@ -0,0 +1,526 @@
+            String state0 = readFile(path0);
+            String className = fieldName.substring(0, fieldName.lastIndexOf("."));
+            String subFieldName = fieldName.substring(fieldName.lastIndexOf(".")+1, fieldName.length());
+
+            Object ob_0;
+            XStream xstream = getXStreamInstance();
+
+            try {
+                Class c = Class.forName(className);
+                Field[] fieldList = c.getDeclaredFields();
+                for (int i=0; i< fieldList.length; i++) {
+                    if (!fieldList[i].getName().equals(subFieldName)) {
+                        continue;
+                    }
+                    try {
+                        fieldList[i].setAccessible(true);
+                        Field modifiersField = Field.class.getDeclaredField("modifiers");
+                        modifiersField.setAccessible(true);
+                        modifiersField.setInt(fieldList[i], fieldList[i].getModifiers() & ~Modifier.FINAL);
+                    }
+                    catch (Exception e) {
+                        String outputPrivateError = fieldName + " reflectionError: " + e + "\n";
+                        Files.write(Paths.get(reflectionFile), outputPrivateError.getBytes(),
+                                StandardOpenOption.APPEND);
+                    }
+                    try {
+                        ob_0 = fieldList[i].get(null);
+                        boolean threadLocal = false;
+                        if (ob_0 instanceof ThreadLocal) {
+                            threadLocal = true;
+                        }
+                        try {
+                            // directly invoke reset when dealing with a Mockito mock object
+                            if (Mockito.mockingDetails(ob_0).isMock()) {
+                                Method m = Mockito.class.getDeclaredMethod("reset", Object[].class);
+                                m.invoke(null, new Object[]{new Object[]{ob_0}});
+                            } else {
+                                UnmarshalChain.reset();
+                                UnmarshalChain.initializeChain(fieldList[i].getDeclaringClass().getName(), fieldList[i].getName());
+                                ob_0 = xstream.fromXML(state0);
+                            }
+                        } catch (NoSuchMethodError nsme) {
+                            UnmarshalChain.reset();
+                            UnmarshalChain.initializeChain(fieldList[i].getDeclaringClass().getName(), fieldList[i].getName());
+                            ob_0 = xstream.fromXML(state0);
+                        }
+                        try {
+                            if (AccessibleObject.class.isAssignableFrom(ob_0.getClass())) {
+                                ((AccessibleObject) ob_0).setAccessible(true);
+                            }
+                        } catch (NullPointerException npex) {
```

Comment:
Is `NullPointerException` really the only possible exception here? I assume we can just check for null (and figure out why it can be null...).